plugins/knot/knot

```diff
@@ -112,14 +113,26 @@ def get_stats():
     """Get statistics."""
     # Get status output
     try:
-        pipe = subprocess.Popen(
-            ['/usr/sbin/knotc', '--force', 'stats'],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT)
-        output = pipe.communicate()[0].decode('utf-8', 'ignore')
-    except OSError:
+        output = subprocess.run(['knotc', '--force', 'stats'],
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE, check=False,
+                                encoding='utf-8', errors='ignore').stdout
+    except FileNotFoundError:
         return {}
 
+    # After server reboot output can be almost empty. Use cached results
+    # instead, needed for plugin config when using munin-async.
+    cachename = os.getenv('MUNIN_PLUGSTATE') + '/knot.state'
```

Comment:
`os.path.join(...)` would be better for non-posix platforms, I guess.

Reply:
I wonder if it supports non-posix: "Knot DNS itself is written in a portable way and can be compiled and run on most UNIX-like systems, such as Linux, *BSD, and macOS."

But changed anyway....